solr/core/src/java/org/apache/solr/core/SolrPaths.java

```diff
@@ -43,9 +42,7 @@ private SolrPaths() {} // don't create this
 
   /** Ensures a directory name always ends with a '/'. */
   public static String normalizeDir(String path) {
-    return (path != null && (!(path.endsWith("/") || path.endsWith("\\"))))
-        ? path + File.separator
-        : path;
+    return (path != null && (!(path.endsWith("/") || path.endsWith("\\")))) ? path + "/" : path;
```

Comment:
wow that's gnarly..   Is there no existing `normalize()` function in any of our jars that would do this for us?   I guess fixing that is orthongonal to this effort....